snippets/javascript/password-rule-checker.md

```diff
@@ -0,0 +1,161 @@
+# Password Rule Checker
+
+Demo [click here](https://passwordrulechecker.glitch.me/)
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Passoword Rule Checker</title>
+    <!-- Bootstrap CSS-->
+    <link
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css"
+      rel="stylesheet"
+      integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3"
+      crossorigin="anonymous"
+    />
+  </head>
+  <body>
+    <div class="container mt-5">
+      <div class="row">
+        <div class="col-md-3"></div>
+        <div class="col-md-6">
+          <form>
+            <div class="form-group">
+              <label for="email">Email address</label>
+              <input
+                type="email"
+                value="sample@email.com"
+                class="form-control"
+                id="email"
+                readonly
+              />
+            </div>
+            <div class="form-group">
+              <label for="password">Password</label>
+              <input
+                type="password"
+                class="form-control"
+                id="password"
+                autocomplete="off"
+                placeholder="Enter your password"
+              />
+            </div>
+            <button type="submit" class="btn btn-primary m-3">Submit</button>
+            <hr />
+            <progress
+              min="0"
+              max="100"
+              value="0"
+              class="form-control"
+              id="progressbar"
+            >
+              Password
+            </progress>
+            <hr />
+            <div>
+              <small>PASSWORD SHOULD</small><br />
```

Comment:
Something cool to include if you like a challenge 😎 (or just to think about how you would implement this) : <br/>
Include a list/array with the 100 most used passwords (https://solvit.io/a86f047) which you use to disalow weak passwords.